### PR TITLE
[linux] Enable setup code options for linux example apps

### LINKED ISSUE #8404: Enhance Lighting example app to use user provided QRcode or setup code
Current lighting app using hardcoded DeviceConfiguration information, instead it should be enhanced to take input QRcode/setup code, this will be helpful for CSG tests validation. 

also Lighting App needs to implement basic information cluster.

### PR DESCRIPTION
**Pull request by yufengwangca**
#### Problem
What is being fixed?  Examples:
* currently, example app always generate setup code with default Device Configuration, CSG would like to add options to allow configure QRcode/setup code with linux example apps for CSG tests validation.

* Fixes #8404 

#### Change overview
Enable setup code options for linux example apps

#### Testing
How was this tested? (at least one bullet point required)
1. Run chip-lighting-app -h to show new options for Setup Payload.
```
yufengw@yufengw-SEi:~/Workspace/connectedhomeip/examples/lighting-app/linux/out/debug$ ./chip-lighting-app -h
Usage: ./chip-lighting-app [options]

GENERAL OPTIONS

  --ble-device <number>
       The device number for CHIPoBLE, without 'hci' prefix, can be found by hciconfig.

  --wifi
       Enable WiFi management via wpa_supplicant.

  --thread
       Enable Thread management via ot-agent.

  --version <version>
       The version indication provides versioning of the setup payload.

  --vendor-id <id>
       The Vendor ID is assigned by the Connectivity Standards Alliance.

  --product-id <id>
       The Product ID is specified by vendor.

  --custom-flow <Standard = 0 | UserActionRequired = 1 | Custom = 2>
       A 2-bit unsigned enumeration specifying manufacturer-specific custom flow options.

  --capabilities <None = 0, SoftAP = 1 << 0, BLE = 1 << 1, OnNetwork = 1 << 2>
       Discovery Capabilities Bitmask which contains information about Device’s available technologies for device discovery.

  --discriminator <discriminator>
       A 12-bit unsigned integer match the value which a device advertises during commissioning.

  --passcode <passcode>
       A 27-bit unsigned integer, which serves as proof of possession during commissioning.

HELP OPTIONS

  -h, --help
       Print this output and then exit.

  -v, --version
       Print the version and then exit.
```
2. Run ./chip-lighting-app with customized setup payload  
```
./chip-lighting-app --version 1 --vendor-id 1234 --product-id 1234 --custom-flow 2 --capabilities 3 --discriminator 1234 --passcode 1234 

[1626884897.552607][453003:453003] CHIP:SVR: SetupQRCode: [MT:TA6K4-J-02MOJB00000]
[1626884897.552637][453003:453003] CHIP:SVR: Manual pairing code: [501234000001234012341]
```

### CODE REVIEW

**Comment by yufengwangca** [comment]
@bhaskar-apple @msandstedt @jepenven-silabs @mspang ?

File: src/app/server/OnboardingCodesUtil.cpp [hunk 1]
```diff
@@ -67,6 +67,39 @@ void PrintOnboardingCodes(chip::RendezvousInformationFlags aRendezvousFlags)
     }
 }
 
+void PrintOnboardingCodes(const chip::SetupPayload & payload)
+{
+    std::string QRCode;
```
Comment (mspang) [inline src/app/server/OnboardingCodesUtil.cpp:72]: nit: `qrCode`

**Review by mspang — APPROVED** [review]

File: src/app/server/OnboardingCodesUtil.cpp (hunk 1 above)
Comment (yufengwangca) [reply src/app/server/OnboardingCodesUtil.cpp:72]: I was thinking of it, but still keep this name to align with other places in the current code base

**Review by msandstedt — APPROVED** [review]

**Comment by yufengwangca** [comment]
@bhaskar-apple ?

File: src/app/server/OnboardingCodesUtil.cpp [hunk 2]
```diff
@@ -36,17 +36,17 @@ using namespace ::chip::DeviceLayer;
...
     {
         chip::Platform::ScopedMemoryBuffer<char> qrCodeBuffer;
-        const size_t qrCodeBufferMaxSize = strlen(kQrCodeBaseUrl) + strlen(kUrlDataAssignmentPhrase) + 3 * QRCode.size() + 1;
+        const size_t qrCodeBufferMaxSize = strlen(kQrCodeBaseUrl) + strlen(kUrlDataAssignmentPhrase) + 3 * qrCode.size() + 1;
         qrCodeBuffer.Alloc(qrCodeBufferMaxSize);
 
-        ChipLogProgress(AppServer, "SetupQRCode: [%s]", QRCode.c_str());
-        if (GetQRCodeUrl(&qrCodeBuffer[0], qrCodeBufferMaxSize, QRCode) == CHIP_NO_ERROR)
+        ChipLogProgress(AppServer, "SetupQRCode: [%s]", qrCode.c_str());
+        if (GetQRCodeUrl(&qrCodeBuffer[0], qrCodeBufferMaxSize, qrCode) == CHIP_NO_ERROR)
```
Comment (bzbarsky-apple) [inline src/app/server/OnboardingCodesUtil.cpp:49]:
```suggestion
        if (GetQRCodeUrl(qrCodeBuffer.Get(), qrCodeBufferMaxSize, qrCode) == CHIP_NO_ERROR)
```

File: src/app/server/OnboardingCodesUtil.cpp [hunk 3]
```diff
@@ -36,17 +36,17 @@ using namespace ::chip::DeviceLayer;
...
         {
             ChipLogProgress(AppServer, "Copy/paste the below URL in a browser to see the QR Code:");
             ChipLogProgress(AppServer, "%s", &qrCodeBuffer[0]);
```
Comment (bzbarsky-apple) [inline src/app/server/OnboardingCodesUtil.cpp:52]:
```suggestion
            ChipLogProgress(AppServer, "%s", qrCodeBuffer.Get());
```

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
> @bhaskar-apple ?

Not active on Matter anymore.

**Review by saurabhst — APPROVED** [review]

**Pull request merged**